File: Src/Graphology.TestsF/GraphNodeTests.fs
namespace TeaDriven.Graphology.Tests

open Xunit
open Xunit.Extensions
open Ploeh.AutoFixture
open Ploeh.AutoFixture.Idioms
open FsUnit.Xunit

module GraphNodeTests =
    open TeaDriven.Graphology
    open TeaDriven.Graphology.Tests.Extensions

    [<Theory; AutoFoqData>]
    let ``Constructors should have null guards`` (assertion : GuardClauseAssertion) =
        assertion.Verify (typeof<GraphNode>.GetConstructors())

    [<Theory; AutoFoqData>]
    let ``Properties should be writable`` (assertion : WritablePropertyAssertion, fixture : IFixture) =
        fixture.MakeNonRecursive()
        assertion.Verify typeof<GraphNode>

    [<Theory; AutoFoqData>]
    let ``Subgraph should be empty when not set`` (fixture : IFixture) =
        fixture.MakeNonRecursive()
        fixture.Customize(new NoAutoPropertiesCustomization(typeof<GraphNode>))

        let sut = fixture.Create<GraphNode>()

        Assert.Empty(sut.SubGraph)

File: Src/TeaDriven.Graphology.Tests/GraphNodeTests.cs
using Ploeh.AutoFixture;
using Ploeh.AutoFixture.Idioms;
using TeaDriven.Graphology.Tests.Conventions;
using Xunit;
using Xunit.Extensions;

namespace TeaDriven.Graphology.Tests
{
    public class GraphNodeTests
    {
        [Theory, AutoNSubstituteData]
        public void Constructor_HasNullGuards(GuardClauseAssertion assertion)
        {
            assertion.Verify(typeof(GraphNode).GetConstructors());
        }

        [Theory, AutoNSubstituteData]
        public void Properties_AreWritable(WritablePropertyAssertion assertion, IFixture fixture)
        {
            fixture.MakeNonRecursive();

            assertion.Verify(typeof(GraphNode));
        }

        [Theory, AutoNSubstituteData]
        public void SubGraph_NotSet_ReturnsEmptyList(IFixture fixture)
        {
            // Arrange
            fixture.MakeNonRecursive();

            fixture.Customize(new NoAutoPropertiesCustomization(typeof(GraphNode)));

            var sut = fixture.Create<GraphNode>();

            // Act
            // Assert
            Assert.Empty(sut.SubGraph);
        }
    }
}